javaagent-bootstrap/javaagent-bootstrap.gradle

```diff
@@ -7,6 +7,18 @@ apply from: "$rootDir/gradle/publish.gradle"
 minimumBranchCoverage = 0.0
 minimumInstructionCoverage = 0.0
 
+// patch inner class from Caffeine to avoid ForkJoinTask from being loaded too early
+sourceSets {
+  patch {
+    java {}
+  }
+}
+jar {
+  from(sourceSets.patch.output) {
+    include 'com/github/benmanes/caffeine/cache/BoundedLocalCache$PerformCleanupTask.class'
```

Comment:
Will this file be shaded too? Also, why `javaagent-bootstrap` and not the new module?

Reply:
Oh good catch, I don't think this is doing anything because of the non shading (I'll need to come up with a bespoke test for this I guess). It's here since only the agent cares, not library instrumentation.

Reply:
For additional reference there is a gap here vs datadog - they keep Caffeine in the agent classloader but we are shading and putting in the bootstrap classloader instead to allow the same instrumentation to work for both library and agent.

Reply:
I fixed the package name but couldn't come up with a way to add a test for this. Anyone have any ideas?

Reply:
Was able to come up with a test, a bit annoying since I need a different Gradle project to be able to check the patched jar itself.